specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1006 @@
+
+Constants of any type are permitted as part of a function call. What data types
+occur in the program hence depends on what QIS functions are used in addition to
+the runtime functions for initialization and output recording. Constant values
+of type `i64` in particular may be used as part of calls to output recording
+functions; see the section on [output recording](#output-recording) for more
+details.
+
+The `%Qubit*` and `%Result*` data types must be supported by all backends.
+Qubits and results can occur only as arguments in function calls and are
+represented as a pointer of type `%Qubit*` and `%Result*` respectively, where
+the pointer itself identifies the qubit or result value rather than a memory
+location where the value is stored: a 64-bit integer constant is cast to the
+appropriate pointer type. A more detailed elaboration on the purpose of this
+representation is given in the next subsection. The integer constant that is
+cast must be in the interval `[0, numQubits)` for `%Qubit*` and `[0,
+numResults)` for `%Result*`, where `numQubits` and `numResults` are the required
+number of qubits and results defined by the corresponding [entry point
+attributes](#attributes). Since backends may look at the values of the
+`required_num_qubits` and `required_num_results` attributes to determine whether
+a program can be executed, it is recommended to index qubits and results
+consecutively so that there are no unused values within these ranges.
+
+### Qubit and Result Usage
+
+Qubits and result values are represented as opaque pointers in the bitcode,
+which may only ever be dereferenced as part of a runtime function
+implementation. In general, the QIR specification distinguishes between two
+kinds of pointers for representing a qubit or result value, as explained in more
+detail [here](../Execution.md), and either one, though not both, may be used
+throughout a bitcode file. A [module flag](#module-flags-metadata) in the
+bitcode indicates which kinds of pointers are used to represent qubits and
+result values.
+
+The first kind of pointer points to a valid memory location that is managed
+dynamically during program execution, meaning the necessary memory is allocated
+and freed by the runtime. The second kind of pointer merely identifies a qubit
+or result value by a constant integer encoded in the pointer itself. To be
+compliant with the Adaptive Profile specification, the program must not make use
+of dynamic qubit or result management, meaning it must use only the second kind
+of pointer; qubits and results must be identified by a constant integer value
+that is bitcast to a pointer to match the expected type. How such an integer
+value is interpreted and specifically how it relates to hardware resources is
+ultimately up to the executing backend.
+
+Additionally, the Adaptive Profile imposes the following restrictions on qubit
+and result usage:
+
+- Qubits must not be used after they have been passed as arguments to a function
```

Comment:
This seems inconsistent with the rest of the specification. Up to this point, qubit reuse is a required capability, but this point conveys the message that qubit reuse is not supported in an adaptive profile program.